src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -68,35 +92,37 @@ public AttachCaseCallbackService(CcdApi ccdApi, PaymentsProcessor paymentsProces
      * @return Either exception record field map, when processing was successful,
      *         or the list of errors, in case of errors
      */
-    public Either<List<String>, Map<String, Object>> process(
-        CaseDetails exceptionRecord,
+    public Either<ErrorsAndWarnings, Map<String, Object>> process(
+        CaseDetails exceptionRecordDetails,
         String requesterIdamToken,
         String requesterUserId,
-        String eventId
+        String eventId,
+        Boolean ignoreWarnings
     ) {
         Validation<String, Void> eventIdValidation = isAttachToCaseEvent(eventId);
 
         if (eventIdValidation.isInvalid()) {
             String eventIdValidationError = eventIdValidation.getError();
             log.warn("Validation error {}", eventIdValidationError);
-            return Either.left(singletonList(eventIdValidationError));
+            return Either.left(ErrorsAndWarnings.withErrors(singletonList(eventIdValidationError)));
         }
 
-        Validation<String, Void> classificationValidation = canBeAttachedToCase(exceptionRecord);
+        Validation<String, Void> classificationValidation = canBeAttachedToCase(exceptionRecordDetails);
 
         if (classificationValidation.isInvalid()) {
             String eventIdClassificationValidationError = classificationValidation.getError();
             log.warn("Validation error {}", eventIdClassificationValidationError);
-            return Either.left(singletonList(eventIdClassificationValidationError));
+            return Either.left(ErrorsAndWarnings.withErrors(singletonList(eventIdClassificationValidationError)));
         }
-        boolean useSearchCaseReference = isSearchCaseReferenceTypePresent(exceptionRecord);
+        boolean useSearchCaseReference = isSearchCaseReferenceTypePresent(exceptionRecordDetails);
 
-        return getValidation(exceptionRecord, useSearchCaseReference, requesterIdamToken, requesterUserId)
-            .map(event -> tryAttachToCase(event, exceptionRecord))
+        return getValidation(exceptionRecordDetails, useSearchCaseReference, requesterIdamToken, requesterUserId)
+            .map(callBackEvent -> tryAttachToCase(callBackEvent, exceptionRecordDetails, ignoreWarnings))
             .map(attachCaseResult ->
-                attachCaseResult.map(modifiedFields -> mergeCaseFields(exceptionRecord.getData(), modifiedFields))
+                attachCaseResult.map(modifiedFields ->
+                    mergeCaseFields(exceptionRecordDetails.getData(), modifiedFields))
             )
-            .getOrElseGet(errors -> Either.left(errors.toJavaList()));
+            .getOrElseGet(errors -> Either.left(ErrorsAndWarnings.withErrors(errors.toJavaList())));
     }
 
     private Validation<Seq<String>, AttachToCaseEventData> getValidation(
```

Comment:
This bit has grown so big that it would make sense to extract it to a separate class. Doesn't have to be done in this PR.